File: src/main/java/no/fdk/dataset_catalog/Application.java
package no.fdk.dataset_catalog;

import no.fdk.dataset_catalog.configuration.ApplicationProperties;
import no.fdk.dataset_catalog.configuration.SecurityProperties;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication
@EnableConfigurationProperties({ApplicationProperties.class, SecurityProperties.class})
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: src/main/kotlin/no/fdk/dataset_catalog/Application.kt
package no.fdk.dataset_catalog

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import no.fdk.dataset_catalog.configuration.ApplicationProperties
import no.fdk.dataset_catalog.configuration.SecurityProperties
import org.springframework.boot.SpringApplication

@SpringBootApplication
@EnableConfigurationProperties(ApplicationProperties::class, SecurityProperties::class)
open class Application

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}
